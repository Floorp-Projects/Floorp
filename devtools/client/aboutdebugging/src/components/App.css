/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

/*
 * The current layout of about:debugging is
 *
 *  +-------------+-------------------------------+
 *  |   Sidebar   | Page (Runtime or Connect)     |
 *  |   (240px)   |                               |
 *  |             |                               |
 *  +-------------+-------------------------------+
 *
 * Some of the values (font sizes, widths, etc.) are the same as
 * about:preferences, which uses the shared common.css
 */

.app {
  /* from common */
  --sidebar-width: 280px;
  --app-top-padding: 70px;
  --app-bottom-padding: 40px;
  --app-left-padding: 32px;
  --app-right-padding: 32px;

  box-sizing: border-box;
  width: 100vw;
  height: 100vh;
  overflow: hidden; /* we don't want the sidebar to scroll, only the main content */

  display: grid;
  grid-column-gap: 40px;
  grid-template-columns: var(--sidebar-width) auto;

  font-size: var(--base-font-size);
  font-weight: var(--base-font-weight);
  line-height: var(--base-line-height);
}

.app__sidebar {
  padding-block-start: var(--app-top-padding);
  padding-block-end: var(--app-bottom-padding);
  padding-inline-start: var(--app-left-padding);
}

.app__content {
  /* we want to scroll only the main content, not the sidebar */
  overflow-y: auto;

  /* padding will give space for card shadow to appear and
     margin will correct the alignment */
  margin-inline-start: calc(var(--card-shadow-blur-radius) * -1);
  padding-inline: var(--card-shadow-blur-radius);
  padding-block-start: var(--app-top-padding);
}

/* Workaround for Gecko clipping the padding-bottom of a scrollable container;
   we create a block to act as the bottom padding instead. */
.app__content::after {
  content: "";
  display: block;
  height: var(--app-bottom-padding);
}

.page {
  max-width: var(--page-width);
  min-width: min-content;
  font-size: var(--body-20-font-size);
  font-weight: var(--body-20-font-weight);
  padding-inline-end: var(--app-right-padding);
}
