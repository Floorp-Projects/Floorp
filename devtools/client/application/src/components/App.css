/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

/*
 * General styles
 */

a.disabled-link,
a.disabled-link:hover,
a.disabled-link:visited {
  opacity: 0.5;
  cursor: not-allowed;
}

.app {
  display: grid;
  min-width: calc(var(--base-unit) * 90);
}

/* wide layout -> two columns, 1 row */
@media(min-width: 701px) {
  .app {
    grid-template-columns: calc(var(--base-unit) * 50) auto;
    height: 100vh;
  }
}

/* vertical layout -> 1 column, 2 rows */
@media(max-width: 700px) {
  .app {
    grid-template-rows: auto 1fr;
  }
}
