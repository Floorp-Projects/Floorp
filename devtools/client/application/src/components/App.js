/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const PropTypes = require("devtools/client/shared/vendor/react-prop-types");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const { main } = require("devtools/client/shared/vendor/react-dom-factories");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const LocalizationProvider = createFactory(FluentReact.LocalizationProvider);

const PageSwitcher = createFactory(
  require("devtools/client/application/src/components/routing/PageSwitcher")
);
const Sidebar = createFactory(
  require("devtools/client/application/src/components/routing/Sidebar")
);

/**
 * This is the main component for the application panel.
 */
class App extends PureComponent {
  static get propTypes() {
    return {
      fluentBundles: PropTypes.array.isRequired,
    };
  }

  render() {
    const { fluentBundles } = this.props;

    return LocalizationProvider(
      { bundles: fluentBundles },
      main({ className: `app` }, Sidebar({}), PageSwitcher({}))
    );
  }
}

module.exports = App;
