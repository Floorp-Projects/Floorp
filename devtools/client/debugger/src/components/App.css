/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

* {
  box-sizing: border-box;
}

html,
body {
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;
}

#mount {
  height: 100%;
}

button {
  background: transparent;
  border: none;
  font-family: inherit;
  font-size: inherit;
}

button:hover,
button:focus {
  background-color: var(--theme-toolbar-background-hover);
}

.theme-dark button:hover,
.theme-dark button:focus {
  background-color: var(--theme-toolbar-hover);
}

.debugger {
  display: flex;
  flex: 1;
  height: 100%;
}

.debugger .tree-indent {
  width: 16px;
  margin-inline-start: 0;
  border-inline-start: 0;
}

.editor-pane {
  display: flex;
  position: relative;
  flex: 1;
  background-color: var(--theme-body-background);
  height: 100%;
  overflow: hidden;
}

.editor-container {
  width: 100%;
}

/* Utils */
.absolute-center {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.d-flex {
  display: flex;
}

.align-items-center {
  align-items: center;
}

.rounded-circle {
  border-radius: 50%;
}

.text-white {
  color: white;
}

.text-center {
  text-align: center;
}

.min-width-0 {
  min-width: 0;
}

/*
  Prevents horizontal scrollbar from displaying when
  right pane collapsed (#7505)
*/
.split-box > .splitter:last-child {
  display: none;
}

/**
 * In RTL layouts, the Debugger UI overlays the splitters. See Bug 1731233.
 * Note: we need to the `.debugger` prefix here to beat the specificity of the
 * general rule defined in SlitBox.css for `.split-box.vert > .splitter`.
 */
.debugger .split-box.vert > .splitter {
  border-left-width: var(--devtools-splitter-inline-start-width);
  border-right-width: var(--devtools-splitter-inline-end-width);

  margin-left: calc(-1 * var(--devtools-splitter-inline-start-width) - 1px);
  margin-right: calc(-1 * var(--devtools-splitter-inline-end-width));
}

::-webkit-scrollbar {
  width: 8px;
  height: 8px;
  background: transparent;
}

::-webkit-scrollbar-track {
  border-radius: 8px;
  background: transparent;
}

::-webkit-scrollbar-thumb {
  border-radius: 8px;
  background: rgba(113, 113, 113, 0.5);
}
