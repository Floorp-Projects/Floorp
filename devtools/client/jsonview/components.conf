# -*- Mode: python; indent-tabs-mode: nil; tab-width: 40 -*-
# vim: set filetype=python:
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

JSON_VIEW_MIME_TYPE = "application/vnd.mozilla.json.view"

Classes = [
    {
        'cid': '{4148c488-dca1-49fc-a621-2a0097a62422}',
        'contract_ids': ['@mozilla.org/devtools/jsonview-sniffer;1'],
        'jsm': 'resource://devtools/client/jsonview/Sniffer.jsm',
        'constructor': 'Sniffer',
        'categories': {
            'net-content-sniffers': 'JSONView',
            'net-and-orb-content-sniffers': 'JSONView',
         },
    },
    {
        'cid': '{d8c9acee-dec5-11e4-8c75-1681e6b88ec1}',
        'contract_ids': ['@mozilla.org/streamconv;1?from=%s&to=*/*' % JSON_VIEW_MIME_TYPE],
        'jsm': 'resource://devtools/client/jsonview/Converter.jsm',
        'constructor': 'Converter',
    },
]
