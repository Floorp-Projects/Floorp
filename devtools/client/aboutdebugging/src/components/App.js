/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

const Route = createFactory(
  require("devtools/client/shared/vendor/react-router-dom").Route
);
const Switch = createFactory(
  require("devtools/client/shared/vendor/react-router-dom").Switch
);
const Redirect = createFactory(
  require("devtools/client/shared/vendor/react-router-dom").Redirect
);

const Types = require("devtools/client/aboutdebugging/src/types/index");
const {
  PAGE_TYPES,
  RUNTIMES,
} = require("devtools/client/aboutdebugging/src/constants");

const ConnectPage = createFactory(
  require("devtools/client/aboutdebugging/src/components/connect/ConnectPage")
);
const RuntimePage = createFactory(
  require("devtools/client/aboutdebugging/src/components/RuntimePage")
);
const Sidebar = createFactory(
  require("devtools/client/aboutdebugging/src/components/sidebar/Sidebar")
);

class App extends PureComponent {
  static get propTypes() {
    return {
      adbAddonStatus: Types.adbAddonStatus,
      // The "dispatch" helper is forwarded to the App component via connect.
      // From that point, components are responsible for forwarding the dispatch
      // property to all components who need to dispatch actions.
      dispatch: PropTypes.func.isRequired,
      // getString prop is injected by the withLocalization wrapper
      getString: PropTypes.func.isRequired,
      isAdbReady: PropTypes.bool.isRequired,
      isScanningUsb: PropTypes.bool.isRequired,
      networkLocations: PropTypes.arrayOf(Types.location).isRequired,
      networkRuntimes: PropTypes.arrayOf(Types.runtime).isRequired,
      selectedPage: Types.page,
      selectedRuntimeId: PropTypes.string,
      usbRuntimes: PropTypes.arrayOf(Types.runtime).isRequired,
    };
  }

  componentDidUpdate() {
    this.updateTitle();
  }

  updateTitle() {
    const { getString, selectedPage, selectedRuntimeId } = this.props;

    const pageTitle =
      selectedPage === PAGE_TYPES.RUNTIME
        ? getString("about-debugging-page-title-runtime-page", {
            selectedRuntimeId,
          })
        : getString("about-debugging-page-title-setup-page");

    document.title = pageTitle;
  }

  renderConnect() {
    const { adbAddonStatus, dispatch, networkLocations } = this.props;

    return ConnectPage({
      adbAddonStatus,
      dispatch,
      networkLocations,
    });
  }

  // The `match` object here is passed automatically by the Route object.
  // We are using it to read the route path.
  // See react-router docs:
  // https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/match.md
  renderRuntime({ match }) {
    const isRuntimeAvailable = id => {
      const runtimes = [
        ...this.props.networkRuntimes,
        ...this.props.usbRuntimes,
      ];
      const runtime = runtimes.find(x => x.id === id);
      return runtime?.runtimeDetails;
    };

    const { dispatch } = this.props;

    let runtimeId = match.params.runtimeId || RUNTIMES.THIS_FIREFOX;
    if (match.params.runtimeId !== RUNTIMES.THIS_FIREFOX) {
      const rawId = decodeURIComponent(match.params.runtimeId);
      if (isRuntimeAvailable(rawId)) {
        runtimeId = rawId;
      } else {
        // Also redirect to "This Firefox" if runtime is not found
        return Redirect({ to: `/runtime/${RUNTIMES.THIS_FIREFOX}` });
      }
    }

    // we need to pass a key so the component updates when we want to showcase
    // a different runtime
    return RuntimePage({ dispatch, key: runtimeId, runtimeId });
  }

  renderRoutes() {
    return Switch(
      {},
      Route({
        path: "/setup",
        render: () => this.renderConnect(),
      }),
      Route({
        path: "/runtime/:runtimeId",
        render: routeProps => this.renderRuntime(routeProps),
      }),
      // default route when there's no match which includes "/"
      // TODO: the url does not match "/" means invalid URL,
      // in this case maybe we'd like to do something else than a redirect.
      // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1509897
      Route({
        render: routeProps => {
          const { pathname } = routeProps.location;
          // The old about:debugging supported the following routes:
          // about:debugging#workers, about:debugging#addons and about:debugging#tabs.
          // Such links can still be found in external documentation pages.
          // We redirect to This Firefox rather than the Setup Page here.
          if (
            pathname === "/workers" ||
            pathname === "/addons" ||
            pathname === "/tabs"
          ) {
            return Redirect({ to: `/runtime/${RUNTIMES.THIS_FIREFOX}` });
          }
          return Redirect({ to: "/setup" });
        },
      })
    );
  }

  render() {
    const {
      adbAddonStatus,
      dispatch,
      isAdbReady,
      isScanningUsb,
      networkRuntimes,
      selectedPage,
      selectedRuntimeId,
      usbRuntimes,
    } = this.props;

    return Localized(
      {},
      dom.div(
        { className: "app" },
        Sidebar({
          adbAddonStatus,
          className: "app__sidebar",
          dispatch,
          isAdbReady,
          isScanningUsb,
          networkRuntimes,
          selectedPage,
          selectedRuntimeId,
          usbRuntimes,
        }),
        dom.main({ className: "app__content" }, this.renderRoutes())
      )
    );
  }
}

const mapStateToProps = state => {
  return {
    adbAddonStatus: state.ui.adbAddonStatus,
    isAdbReady: state.ui.isAdbReady,
    isScanningUsb: state.ui.isScanningUsb,
    networkLocations: state.ui.networkLocations,
    networkRuntimes: state.runtimes.networkRuntimes,
    selectedPage: state.ui.selectedPage,
    selectedRuntimeId: state.runtimes.selectedRuntimeId,
    usbRuntimes: state.runtimes.usbRuntimes,
  };
};

const mapDispatchToProps = dispatch => ({
  dispatch,
});

module.exports = FluentReact.withLocalization(
  connect(mapStateToProps, mapDispatchToProps)(App)
);
