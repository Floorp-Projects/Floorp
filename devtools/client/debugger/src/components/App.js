/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React, { Component } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import { connect } from "../utils/connect";
import { prefs, features } from "../utils/prefs";
import actions from "../actions";
import A11yIntention from "./A11yIntention";
import { ShortcutsModal } from "./ShortcutsModal";

import {
  getSelectedSource,
  getPaneCollapse,
  getActiveSearch,
  getQuickOpenEnabled,
  getOrientation,
} from "../selectors";

const KeyShortcuts = require("devtools/client/shared/key-shortcuts");
const SplitBox = require("devtools/client/shared/components/splitter/SplitBox");
const AppErrorBoundary = require("devtools/client/shared/components/AppErrorBoundary");

import Services from "devtools-services";
const shortcuts = new KeyShortcuts({ window });

const { appinfo } = Services;

const isMacOS = appinfo.OS === "Darwin";

const horizontalLayoutBreakpoint = window.matchMedia("(min-width: 800px)");
const verticalLayoutBreakpoint = window.matchMedia(
  "(min-width: 10px) and (max-width: 799px)"
);

import "./variables.css";
import "./App.css";

import "./shared/menu.css";

import ProjectSearch from "./ProjectSearch";
import PrimaryPanes from "./PrimaryPanes";
import Editor from "./Editor";
import SecondaryPanes from "./SecondaryPanes";
import WelcomeBox from "./WelcomeBox";
import EditorTabs from "./Editor/Tabs";
import EditorFooter from "./Editor/Footer";
import QuickOpenModal from "./QuickOpenModal";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shortcutsModalEnabled: false,
      startPanelSize: 0,
      endPanelSize: 0,
    };
  }

  getChildContext() {
    return {
      fluentBundles: this.props.fluentBundles,
      toolboxDoc: this.props.toolboxDoc,
      shortcuts,
      l10n: L10N,
    };
  }

  componentDidMount() {
    horizontalLayoutBreakpoint.addListener(this.onLayoutChange);
    verticalLayoutBreakpoint.addListener(this.onLayoutChange);
    this.setOrientation();

    shortcuts.on(L10N.getStr("symbolSearch.search.key2"), e =>
      this.toggleQuickOpenModal(e, "@")
    );

    const searchKeys = [
      L10N.getStr("sources.search.key2"),
      L10N.getStr("sources.search.alt.key"),
    ];
    searchKeys.forEach(key => shortcuts.on(key, this.toggleQuickOpenModal));

    shortcuts.on(L10N.getStr("gotoLineModal.key3"), e =>
      this.toggleQuickOpenModal(e, ":")
    );

    shortcuts.on("Escape", this.onEscape);
    shortcuts.on("Cmd+/", this.onCommandSlash);
  }

  componentWillUnmount() {
    horizontalLayoutBreakpoint.removeListener(this.onLayoutChange);
    verticalLayoutBreakpoint.removeListener(this.onLayoutChange);
    shortcuts.off(
      L10N.getStr("symbolSearch.search.key2"),
      this.toggleQuickOpenModal
    );

    const searchKeys = [
      L10N.getStr("sources.search.key2"),
      L10N.getStr("sources.search.alt.key"),
    ];
    searchKeys.forEach(key => shortcuts.off(key, this.toggleQuickOpenModal));

    shortcuts.off(L10N.getStr("gotoLineModal.key3"), this.toggleQuickOpenModal);

    shortcuts.off("Escape", this.onEscape);
  }

  onEscape = e => {
    const {
      activeSearch,
      closeActiveSearch,
      closeQuickOpen,
      quickOpenEnabled,
    } = this.props;
    const { shortcutsModalEnabled } = this.state;

    if (activeSearch) {
      e.preventDefault();
      closeActiveSearch();
    }

    if (quickOpenEnabled) {
      e.preventDefault();
      closeQuickOpen();
    }

    if (shortcutsModalEnabled) {
      e.preventDefault();
      this.toggleShortcutsModal();
    }
  };

  onCommandSlash = () => {
    this.toggleShortcutsModal();
  };

  isHorizontal() {
    return this.props.orientation === "horizontal";
  }

  toggleQuickOpenModal = (e, query) => {
    const { quickOpenEnabled, openQuickOpen, closeQuickOpen } = this.props;

    e.preventDefault();
    e.stopPropagation();

    if (quickOpenEnabled === true) {
      closeQuickOpen();
      return;
    }

    if (query != null) {
      openQuickOpen(query);
      return;
    }
    openQuickOpen();
  };

  onLayoutChange = () => {
    this.setOrientation();
  };

  setOrientation() {
    // If the orientation does not match (if it is not visible) it will
    // not setOrientation, or if it is the same as before, calling
    // setOrientation will not cause a rerender.
    if (horizontalLayoutBreakpoint.matches) {
      this.props.setOrientation("horizontal");
    } else if (verticalLayoutBreakpoint.matches) {
      this.props.setOrientation("vertical");
    }
  }

  renderEditorPane = () => {
    const { startPanelCollapsed, endPanelCollapsed } = this.props;
    const { endPanelSize, startPanelSize } = this.state;
    const horizontal = this.isHorizontal();

    return (
      <div className="editor-pane">
        <div className="editor-container">
          <EditorTabs
            startPanelCollapsed={startPanelCollapsed}
            endPanelCollapsed={endPanelCollapsed}
            horizontal={horizontal}
          />
          <Editor startPanelSize={startPanelSize} endPanelSize={endPanelSize} />
          {!this.props.selectedSource ? (
            <WelcomeBox
              horizontal={horizontal}
              toggleShortcutsModal={() => this.toggleShortcutsModal()}
            />
          ) : null}
          <EditorFooter horizontal={horizontal} />
          <ProjectSearch />
        </div>
      </div>
    );
  };

  toggleShortcutsModal() {
    this.setState(prevState => ({
      shortcutsModalEnabled: !prevState.shortcutsModalEnabled,
    }));
  }

  // Important so that the tabs chevron updates appropriately when
  // the user resizes the left or right columns
  triggerEditorPaneResize() {
    const editorPane = window.document.querySelector(".editor-pane");
    if (editorPane) {
      editorPane.dispatchEvent(new Event("resizeend"));
    }
  }

  renderLayout = () => {
    const { startPanelCollapsed, endPanelCollapsed } = this.props;
    const horizontal = this.isHorizontal();

    return (
      <SplitBox
        style={{ width: "100vw" }}
        initialSize={prefs.endPanelSize}
        minSize={30}
        maxSize="70%"
        splitterSize={1}
        vert={horizontal}
        onResizeEnd={num => {
          prefs.endPanelSize = num;
          this.triggerEditorPaneResize();
        }}
        startPanel={
          <SplitBox
            style={{ width: "100vw" }}
            initialSize={prefs.startPanelSize}
            minSize={30}
            maxSize="85%"
            splitterSize={1}
            onResizeEnd={num => {
              prefs.startPanelSize = num;
            }}
            startPanelCollapsed={startPanelCollapsed}
            startPanel={<PrimaryPanes horizontal={horizontal} />}
            endPanel={this.renderEditorPane()}
          />
        }
        endPanelControl={true}
        endPanel={<SecondaryPanes horizontal={horizontal} />}
        endPanelCollapsed={endPanelCollapsed}
      />
    );
  };

  renderShortcutsModal() {
    const additionalClass = isMacOS ? "mac" : "";

    if (!features.shortcuts) {
      return;
    }

    return (
      <ShortcutsModal
        additionalClass={additionalClass}
        enabled={this.state.shortcutsModalEnabled}
        handleClose={() => this.toggleShortcutsModal()}
      />
    );
  }

  render() {
    const { quickOpenEnabled } = this.props;
    return (
      <div className={classnames("debugger")}>
        <AppErrorBoundary
          className="app-error-boundary"
          panel={L10N.getStr("ToolboxDebugger.label")}
        >
          <A11yIntention>
            {this.renderLayout()}
            {quickOpenEnabled === true && (
              <QuickOpenModal
                shortcutsModalEnabled={this.state.shortcutsModalEnabled}
                toggleShortcutsModal={() => this.toggleShortcutsModal()}
              />
            )}
            {this.renderShortcutsModal()}
          </A11yIntention>
        </AppErrorBoundary>
      </div>
    );
  }
}

App.childContextTypes = {
  toolboxDoc: PropTypes.object,
  shortcuts: PropTypes.object,
  l10n: PropTypes.object,
  fluentBundles: PropTypes.array,
};

const mapStateToProps = state => ({
  selectedSource: getSelectedSource(state),
  startPanelCollapsed: getPaneCollapse(state, "start"),
  endPanelCollapsed: getPaneCollapse(state, "end"),
  activeSearch: getActiveSearch(state),
  quickOpenEnabled: getQuickOpenEnabled(state),
  orientation: getOrientation(state),
});

export default connect(mapStateToProps, {
  setActiveSearch: actions.setActiveSearch,
  closeActiveSearch: actions.closeActiveSearch,
  closeProjectSearch: actions.closeProjectSearch,
  openQuickOpen: actions.openQuickOpen,
  closeQuickOpen: actions.closeQuickOpen,
  setOrientation: actions.setOrientation,
})(App);
