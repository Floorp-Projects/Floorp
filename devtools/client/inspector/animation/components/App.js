/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  Component,
  createFactory,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");
const { connect } = require("devtools/client/shared/vendor/react-redux");

const AnimationDetailContainer = createFactory(
  require("devtools/client/inspector/animation/components/AnimationDetailContainer")
);
const AnimationListContainer = createFactory(
  require("devtools/client/inspector/animation/components/AnimationListContainer")
);
const AnimationToolbar = createFactory(
  require("devtools/client/inspector/animation/components/AnimationToolbar")
);
const NoAnimationPanel = createFactory(
  require("devtools/client/inspector/animation/components/NoAnimationPanel")
);
const SplitBox = createFactory(
  require("devtools/client/shared/components/splitter/SplitBox")
);

class App extends Component {
  static get propTypes() {
    return {
      addAnimationsCurrentTimeListener: PropTypes.func.isRequired,
      animations: PropTypes.arrayOf(PropTypes.object).isRequired,
      detailVisibility: PropTypes.bool.isRequired,
      direction: PropTypes.string.isRequired,
      dispatch: PropTypes.func.isRequired,
      emitEventForTest: PropTypes.func.isRequired,
      getAnimatedPropertyMap: PropTypes.func.isRequired,
      getAnimationsCurrentTime: PropTypes.func.isRequired,
      getComputedStyle: PropTypes.func.isRequired,
      getNodeFromActor: PropTypes.func.isRequired,
      removeAnimationsCurrentTimeListener: PropTypes.func.isRequired,
      rewindAnimationsCurrentTime: PropTypes.func.isRequired,
      selectAnimation: PropTypes.func.isRequired,
      setAnimationsCurrentTime: PropTypes.func.isRequired,
      setAnimationsPlaybackRate: PropTypes.func.isRequired,
      setAnimationsPlayState: PropTypes.func.isRequired,
      setDetailVisibility: PropTypes.func.isRequired,
      setHighlightedNode: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      simulateAnimation: PropTypes.func.isRequired,
      simulateAnimationForKeyframesProgressBar: PropTypes.func.isRequired,
      timeScale: PropTypes.object.isRequired,
      toggleElementPicker: PropTypes.func.isRequired,
      toggleLockingHighlight: PropTypes.func.isRequired,
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return (
      this.props.animations.length !== 0 || nextProps.animations.length !== 0
    );
  }

  render() {
    const {
      addAnimationsCurrentTimeListener,
      animations,
      detailVisibility,
      direction,
      dispatch,
      emitEventForTest,
      getAnimatedPropertyMap,
      getAnimationsCurrentTime,
      getComputedStyle,
      getNodeFromActor,
      removeAnimationsCurrentTimeListener,
      rewindAnimationsCurrentTime,
      selectAnimation,
      setAnimationsCurrentTime,
      setAnimationsPlaybackRate,
      setAnimationsPlayState,
      setDetailVisibility,
      setHighlightedNode,
      setSelectedNode,
      simulateAnimation,
      simulateAnimationForKeyframesProgressBar,
      timeScale,
      toggleElementPicker,
    } = this.props;

    return dom.div(
      {
        id: "animation-container",
        className: detailVisibility ? "animation-detail-visible" : "",
        tabIndex: -1,
      },
      animations.length
        ? [
            AnimationToolbar({
              addAnimationsCurrentTimeListener,
              animations,
              removeAnimationsCurrentTimeListener,
              rewindAnimationsCurrentTime,
              setAnimationsPlaybackRate,
              setAnimationsPlayState,
              timeScale,
            }),
            SplitBox({
              className: "animation-container-splitter",
              endPanel: AnimationDetailContainer({
                addAnimationsCurrentTimeListener,
                emitEventForTest,
                getAnimatedPropertyMap,
                getAnimationsCurrentTime,
                getComputedStyle,
                removeAnimationsCurrentTimeListener,
                setDetailVisibility,
                simulateAnimation,
                simulateAnimationForKeyframesProgressBar,
                timeScale,
              }),
              endPanelControl: true,
              initialHeight: "50%",
              splitterSize: 1,
              minSize: "30px",
              startPanel: AnimationListContainer({
                addAnimationsCurrentTimeListener,
                animations,
                direction,
                dispatch,
                getAnimatedPropertyMap,
                getNodeFromActor,
                removeAnimationsCurrentTimeListener,
                selectAnimation,
                setAnimationsCurrentTime,
                setHighlightedNode,
                setSelectedNode,
                simulateAnimation,
                timeScale,
              }),
              vert: false,
            }),
          ]
        : NoAnimationPanel({
            toggleElementPicker,
          })
    );
  }
}

const mapStateToProps = state => {
  return {
    animations: state.animations.animations,
    detailVisibility: state.animations.detailVisibility,
    timeScale: state.animations.timeScale,
  };
};

module.exports = connect(mapStateToProps)(App);
