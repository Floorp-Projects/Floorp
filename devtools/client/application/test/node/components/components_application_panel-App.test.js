/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

// Import libs
const { shallow } = require("enzyme");
const { createFactory } = require("react");

// Import & init localization
const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const LocalizationProvider = createFactory(FluentReact.LocalizationProvider);

// Import component
const App = createFactory(
  require("devtools/client/application/src/components/App")
);

describe("App", () => {
  it("renders the expected snapshot", () => {
    const wrapper = shallow(
      LocalizationProvider({ bundles: [] }, App({}))
    ).dive(); // dive to bypass the LocalizationProvider wrapper
    expect(wrapper).toMatchSnapshot();
  });
});
