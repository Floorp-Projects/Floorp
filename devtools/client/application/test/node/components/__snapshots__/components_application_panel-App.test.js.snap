// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the expected snapshot 1`] = `
<LocalizationProvider>
  <main
    className="app"
  >
    <Connect(Sidebar) />
    <Connect(PageSwitcher) />
  </main>
</LocalizationProvider>
`;
